src/sharedring.rs

```diff
@@ -29,34 +30,80 @@ fn round_to_page_size<T>(capacity: usize) -> usize {
     let bytes = crate::ringbuf::channel_bufsize::<T>(capacity);
     let ps = page_size();
     let m = bytes % ps;
-    if m == 0 { bytes } else { bytes + ps - m }
+    if m == 0 {
+        bytes
+    } else {
+        bytes + ps - m
+    }
 }
 
 fn eventfd() -> Result<File, std::io::Error> {
     let x = unsafe { libc::eventfd(0, libc::EFD_CLOEXEC) };
-    if x == -1 { Err(std::io::Error::last_os_error()) }
-    else { Ok(unsafe { File::from_raw_fd(x) }) }
+    if x == -1 {
+        Err(std::io::Error::last_os_error())
+    } else {
+        Ok(unsafe { File::from_raw_fd(x) })
+    }
 }
 
 impl Inner {
     fn new<T>(capacity: usize) -> Result<Self, Error> {
+        Self::with_opts::<T>(capacity, None)
+    }
+
+    fn with_hugetlb<T>(capacity: usize, tlbsize: HugetlbSize) -> Result<Self, Error> {
+        Self::with_opts::<T>(capacity, Some(tlbsize))
```

Comment:
This is a little over-engineered. Inner is not a public struct so you can just have `fn new(capacity: usize, tlbsize: Option<HugetlbSize>)` and skip `with_hugetlb` and `with_opts`.